# HG changeset patch
# User Jan Beich <jbeich@FreeBSD.org>
# Date 1543157580 -7200
# Node ID fbba8450659ff174beac3d1d69e7de277ae432e4
# Parent  0db86656655bb4df54ecb6aa40e3d57e81edfb0b
Bug 1509757 - Don't rely on GNU regex extension for character classes. r=froydnj a=lizzard
$ echo --with-system-nspr --with-system-nss | gnused -e 's/--with-system-nspr\S* *//'
--with-system-nss
$ echo --with-system-nspr --with-system-nss | bsdsed -e 's/--with-system-nspr\S* *//'
sed: 1: "s/--with-system-nspr\S* ...": RE error: trailing backslash (\)

diff --git a/old-configure.in b/old-configure.in
--- a/old-configure.in
+++ b/old-configure.in
@@ -4753,17 +4753,17 @@ HOST_LDFLAGS="$_SUBDIR_HOST_LDFLAGS"
 RC=
 
 # Run the SpiderMonkey 'configure' script.
 dist=$MOZ_BUILD_ROOT/dist
 ac_configure_args="$_SUBDIR_CONFIG_ARGS"
 
 # --with-system-nspr will have been converted into the relevant $NSPR_CFLAGS
 # and $NSPR_LIBS.
-ac_configure_args="`echo $ac_configure_args | sed -e 's/--with-system-nspr\S* *//'`"
+ac_configure_args="`echo $ac_configure_args | sed -e 's/--with-system-nspr[[^[:space:]]]* *//'`"
 
 if test "$_INTL_API" = no; then
     ac_configure_args="$ac_configure_args --without-intl-api"
 fi
 
 if test -n "$NSPR_CFLAGS" -o -n "$NSPR_LIBS"; then
     ac_configure_args="$ac_configure_args --with-nspr-cflags='$NSPR_CFLAGS'"
     ac_configure_args="$ac_configure_args --with-nspr-libs='$NSPR_LIBS'"
